src/Css/Style.php

```diff
@@ -1446,6 +1485,15 @@ public function set_prop(string $prop, $val, bool $important = false, bool $clea
                 $val = self::$_defaults[$prop];
             }
 
+            // Do not resolve CSS custom property values when setting them,
+            // because the required custom property might not have been set yet,
+            // or need to be derived from a parent. Let the compute() method
+            // resolve this when the value is needed. Just set the property now.
+            if (is_string($val) && substr($val, 0, 4) === 'var(') {
+                $this->_props[$prop] = $val;
+                return;
+            }
```

Comment:
`compute_prop` doesn't necessarily run for nothing. It's just that it will re-calculate if the custom property changes. One thing I think helps when looking at the code is consistency in implementation and this is how other properties are handled.

On the other hand, resolving the value when it is retrieved would probably save us a few cpu cycles, primarily because inheritance isn't performed until later. I could be convinced that it's a good idea to skip initial computation for properties that utilize the CSS variables. @Mellthas may have thoughts on this.

Reply:
I would say that a `var()` by definition is depending on another property. This might be on the same style, but most likely this will have to be inherited because its value is defined on `:root` or some other element. 

But I'll have a look at creating a proper dependency map in the meantime.

Reply:
Our current logic does syntax checking and computation as a combined step. I think it makes sense to defer computation to later here. From the spec:

> If a property contains one or more [var()](https://www.w3.org/TR/css-variables-1/#funcdef-var) functions, and those functions are syntactically valid, the entire property’s grammar must be assumed to be valid at parse time. It is only syntax-checked at computed-value time, after var() functions have been [substituted](https://www.w3.org/TR/css-variables-1/#substitute-a-var).

Reply:
The check here also needs to account for `var()` occurring anywhere within the value.

Reply:
I agree, some more thought on dependency handling is probably required. Though honestly when I attempted to trigger an invalid resolution by modifying the value of a custom property in later declarations it always came back the expected value. Still, we should be handling that dependency in case it's needed.

Reply:
Based on Melthias comment above I assume we are (for now) fine with setting the value and returning. Marking resolved as we can follow on dependency handline in another thread,